src/Data/List/Relation/Unary/Any/Properties.agda

```diff
@@ -255,24 +262,41 @@ Any-×⁻ pq with Product.map₂ (Product.map₂ find) (find pq)
 
     (p , q) ∎
 
-  to∘from : ∀ pq → Any-×⁺ {xs = xs} (Any-×⁻ pq) ≡ pq
-  to∘from pq with find pq
-      | (λ (f : (proj₁ (find pq) ≡_) ⋐ _) → map∘find pq {f})
-  ... | (x , x∈xs , pq′) | lem₁
-    with find pq′
-      | (λ (f : (proj₁ (find pq′) ≡_) ⋐ _) → map∘find pq′ {f})
-  ... | (y , y∈ys , p , q) | lem₂
-    rewrite sym $ map-∘ {R = λ x → Any (λ y → P x × Q y) ys}
-                        (λ p → Any.map (λ q → p , q) (lose y∈ys q))
-                        (λ y → subst P y p)
-                        x∈xs
-            = lem₁ _ helper
-    where
-    helper : Any.map (λ q → p , q) (lose y∈ys q) ≡ pq′
-    helper rewrite sym $ map-∘ (λ q → p , q)
-                               (λ y → subst Q y q)
-                               y∈ys
-           = lem₂ _ refl
+  to∘from : ∀ pq → Any-×⁺ (Any-×⁻ pq) ≡ pq
+  to∘from pq =
+    let x , x∈xs , pq′ = find pq
+
+        lem₁ : {f : (x ≡_) ⋐ λ x → Any (λ y → (P x) × (Q y)) ys} →
+               (f-refl : f refl ≡ pq′) → Any.map f _ ≡ pq
+        lem₁ {f} = map∘find pq {f = f}
+
+        y , y∈ys , p , q = find pq′
+
+        lem₂ : {g : (y ≡_) ⋐ λ y → (P x) × (Q y)} →
+               (g-refl : g refl ≡ (p , q)) → Any.map g _ ≡ pq′
```

Comment:
Same is true here. But here this lemma is only used once - so perhaps its short proof can instead be inlined at line 284 below?  And if the 'proof' is indeed `map∘find`, here and below, why doesn't the statement involve `find` ?

[Same guess; I've had to un-reduce some equational proofs LHSs to make the reason of the proof clearer in `agda-categories` - perhaps here too?]

Reply:
Well, as I first indicated, it's possible more refactoring is possible, but my brain ran out of resources...

Reply:
The online g seems a good idea which moreover works... ;-) Not so sure about `lem₁` though...

Reply:
As above, inlining worked like a charm! (So I wonder if it could have already done so back in the original proofs... !?)